### PR TITLE
fix SANs are not set on the generated certificates

### PR DESCRIPTION
**Pull request by delgod**
# Description

Please include a summary of the change. Please also include relevant motivation and context. List any dependencies that are required for this change.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation (not needed)
- [ ] I have added tests that validate the behaviour of the software (no tests for previous code)
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Gmerold — COMMENTED** [review]
Please add description of the PR.

File: src/self_signed_certificates.py [hunk 1]
```diff
@@ -68,23 +68,21 @@ def generate_certificate(
...
-        names = [x509.DNSName(n) for n in alt_names]
+    for extension in csr_object.extensions:
         certificate_builder = certificate_builder.add_extension(
-            x509.SubjectAlternativeName(names),
-            critical=False,
+            extension.value,
+            critical=extension.critical,
         )
-    certificate_builder._version = x509.Version.v1
+    certificate_builder._version = x509.Version.v3
```
Comment (gruyaume) [inline src/self_signed_certificates.py:85]: Is this change related to fixing the SAN issue?

File: tests/unit/test_charm.py [hunk 2]
```diff
@@ -44,6 +44,8 @@ def setUp(self):
     def test_given_configuration_options_are_set_and_unit_is_leader_when_config_changed_then_status_is_active(  # noqa: E501
         self,
     ):
+        assert self.harness.charm is not None
```
Comment (gruyaume) [inline tests/unit/test_charm.py:47]: Please don't add all those checks to the unit tests

File: tests/unit/test_charm.py [hunk 3]
```diff
@@ -44,6 +44,8 @@ def setUp(self):
     def test_given_configuration_options_are_set_and_unit_is_leader_when_config_changed_then_status_is_active(  # noqa: E501
```
Comment (gruyaume) [inline tests/unit/test_charm.py:44]: Please add tests to validate the fix you made

**Review by gruyaume — CHANGES_REQUESTED** [review]

File: tests/unit/test_charm.py (hunk 2 above)
Comment (gruyaume) [reply tests/unit/test_charm.py:47]: You can fix the ops version to `ops==1.5.2` in the requirements.txt file

File: src/self_signed_certificates.py (hunk 1 above)
Comment (delgod) [reply src/self_signed_certificates.py:85]: The extensions we are setting are of type: x509.v3, it is then necessary to set the version to the same. Otherwise, the loading of extensions fails with javax.net.ssl.SSLProtocolException: no more data allowed for version 1 certificate.
